### PR TITLE
Handle deprecated NXgeometry like new geometry classes

### PR DESCRIPTION
**Pull request by SimonHeybrock**

### CODE REVIEW

File: src/scippnexus/nxobject.py [hunk 1]
```diff
@@ -423,6 +424,8 @@ def insert(container, name, obj):
             detector_number = container.coords[detector_number]
         for key, child in self[[NXcylindrical_geometry, NXoff_geometry]].items():
             insert(container, key, child.load_as_array(detector_number=detector_number))
+        for key, child in self[NXgeometry].items():
+            insert(container, key, child[()])
```
Comment (jl-wynen) [inline src/scippnexus/nxobject.py:428]: Why does this use `child[()]` and not `child.load_as_array` like the other geometry classes?

**Review by jl-wynen — COMMENTED** [review]
Got any files where this is used?

File: src/scippnexus/nxobject.py (hunk 1 above)
Comment (SimonHeybrock) [reply src/scippnexus/nxobject.py:428]: Because we are not performing any transformations on it, it is just loaded as a plain `DataGroup`.

**Comment by SimonHeybrock** [comment]
> Got any files where this is used?

There is a test in scippneutron that loads a PG3 file that uses this.

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**